### DISCUSSION TITLE
NGINX Proxy Manager: Cannot create SSL Certificates

### DISCUSSION THREAD

**Discussion by Max-Btd** [tteck/Proxmox#2427] [Help Needed] [answered]
Since an update/reinstallation of NGINX Reverse Proxy Manager in Proxmox, i cannot create new ssl certificates.
Whenever I want to create a certificate, I get an Internal Error

The last version i used was from Oct 23. Do I need to consider anything new?

**Comment by tteck** [comment]
For which Certbot?

**Reply by Max-Btd** [reply]
What do you mean? I just installed the Proxy Manager with your Script. Is there anything else I have to do for the certificates?

**Reply by tteck** [reply, marked as answer]
> 🚨 Since there are hundreds of Certbot instances, it's necessary to install the specific Certbot of your preference.

I think it installs with Duckdns, Cloudflare and Porkbun.

**Reply by Max-Btd** [reply]
Okay, do you happen to have any instructions? I'm a bit lost when it comes to this. Previously I installed the Proxy Manager via your script and was able to create certificates directly without having to do anything else.

**Reply by tteck** [reply]
If that's the case, you should be able to now. Where is your domain residing at?

**Reply by fjaviys** [reply]
something new about that?
I tried to make a DNS Challenge with duckdns writing my duckdns token id but I still get error.
After installing the NPM LXC , I went to shell of that LXC and execute "python3 -m pip install --no-cache-dir cffi certbot certbot-dns-duckdns".
Go to npm webadmin , create a host, try to request new cert with dns challenge, but the error comes:

> Add Let's Encrypt Certificate
> Internal Error
> 
> Error: Command failed: certbot certonly --config "/etc/letsencrypt.ini" --work-dir "/tmp/letsencrypt-lib" --logs-dir "/tmp/letsencrypt-log" --cert-name "npm-6" --agree-tos --email "admin@ha.com" --domains "prueba.aaaa.com" --authenticator dns-duckdns --dns-duckdns-credentials "/etc/letsencrypt/credentials/credentials-6"
> Saving debug log to /tmp/letsencrypt-log/letsencrypt.log
> Encountered exception during recovery: certbot.errors.PluginError: The domain "_acme-challenge.prueba.aaaa.com" is not valid a duckdns subdomain.
> The domain "_acme-challenge.prueba.aaaa.com" is not valid a duckdns subdomain.
> Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /tmp/letsencrypt-log/letsencrypt.log or re-run Certbot with -v for more details.
> 
>     at ChildProcess.exithandler (node:child_process:422:12)
>     at ChildProcess.emit (node:events:517:28)
>     at maybeClose (node:internal/child_process:1098:16)
>     at ChildProcess._handle.onexit (node:internal/child_process:303:5)`
